aten/src/ATen/native/ForeachUtils.h

```diff
@@ -137,7 +137,7 @@ bool can_use_fast_route(TensorList tensors, Scalar scalar) {
 #endif
 }
 
-bool can_use_fast_route(TensorList tensors, ArrayRef<double> scalars) {
+bool can_use_fast_route(TensorList tensors, ArrayRef<Scalar> scalars) {
```

Comment:
Does this need to check if type promotion will happen between `tensors` and `scalars`?

Reply:
Yes. Its [done](https://github.com/pytorch/pytorch/pull/48224/files?file-filters%5B%5D=.h&file-filters%5B%5D=.py#diff-0e978890cde97eede8219604e17b10129dae0f8317a8c14272c39d62b874c831R213) in the next PR of this stack.

Reply:
Hmmm, should i move these logic from the following PR to this one?

Reply:
Hmm... is it possible to split up the next PR in the stack? I haven't actually looked at it yet, but each of the bullet points in the "What was done" section sound standalone

My 2 cents:
- each commit on the stack should stand by itself. If we have to revert the next PR in the stack for some reason, or if this PR gets merged before the next PR, then we are looking at potentially buggy behavior in the intermediate state.
- The next PR in the stack is a whopping +1,835 −930 line diff. The delta is +900. That's pretty large, folks generally have a hard time reviewing anything that's over 300 lines lol.

Reply:
Done. Broke it up into several smaller PRs. Each has its logical idea and leads to next PR.

Reply:
Thank you!